lib/Conversion/HandshakeToFIRRTL/HandshakeToFIRRTL.cpp

```diff
@@ -1199,6 +1200,49 @@ bool HandshakeBuilder::visitHandshake(JoinOp op) {
   return buildJoinLogic(inputs, output);
 }
 
+// Joins all the input control signals and connects the result to a fork like
+// output logic.
```

Comment:
I think it should be made clearer that it is only control logic which is forked. Data logic is forwarded directly between in- and outputs.